clients/app/src/views/components/compositions/FanClubInfo/FanClubJoinButton.tsx

```diff
@@ -125,43 +123,52 @@ const FanClubJoinButton: React.FC<IFanClubInfoProps> = ({ data }) => {
    */
   const onInvitationRejectPress = React.useCallback(() => {
     actionRespondInvitation.onClose();
-    setJoinStatus(undefined);
+    setJoinStatus(FanClubMemberStatusEnum.noStatus);
 
     fanClubMembersStore.setResourceId(data.id);
     fanClubMembersStore.setUpdateFieldValue('type', 'reject-invitation');
     fanClubMembersStore.updateData();
   }, [actionRespondInvitation, data.id, fanClubMembersStore]);
 
+  /**
+   * On invitation close button press.
+   *
+   * Action will called from confirmation action-sheet containing three buttons:
+   * - Accept invitation
+   * - Reject invitation
+   * - Close
+   */
+  const onInvitationClosePress = React.useCallback(() => {
+    actionRespondInvitation.onClose();
+  }, [actionRespondInvitation]);
+
   React.useEffect(() => {
     if (storeAdd.status === 'error') {
-      setJoinStatus(undefined);
+      setJoinStatus(FanClubMemberStatusEnum.noStatus);
     }
     if (storeDelete.status === 'error') {
-      setJoinStatus(
-        data.privacy === FanClubPrivacyEnum.public
-          ? FanClubMemberStatusEnum.active
-          : FanClubMemberStatusEnum.pendingRequest
-      );
+      if (data.privacy === FanClubPrivacyEnum.public) {
+        setJoinStatus(FanClubMemberStatusEnum.active);
+      } else {
+        setJoinStatus(FanClubMemberStatusEnum.pendingRequest);
+      }
     }
   }, [joinStatus, data.privacy, storeAdd.status, storeDelete.status]);
 
   const { icon, button } = React.useMemo((): ButtonAttributeInterface => {
-    if (!joinStatus) {
-      return buildButtonAttributes(joinStatus, theme.colors, onJoinPress);
+    if (joinStatus === FanClubMemberStatusEnum.active) {
+      return buildButtonAttributes(joinStatus, theme.colors, onLeavePress);
     }
-
-    switch (joinStatus) {
-      case FanClubMemberStatusEnum.active:
-        return buildButtonAttributes(joinStatus, theme.colors, onLeavePress);
-      case FanClubMemberStatusEnum.pendingRequest:
-        return buildButtonAttributes(joinStatus, theme.colors, onCancelRequestPress);
-
-      case FanClubMemberStatusEnum.pendingInvitation:
-        return buildButtonAttributes(joinStatus, theme.colors, onRespondInvitationPress);
-
-      case FanClubMemberStatusEnum.banned:
-        return buildButtonAttributes(joinStatus, theme.colors);
+    if (joinStatus === FanClubMemberStatusEnum.pendingRequest) {
```

Comment:
don't we need else if-s here?

Reply:
each `if` condition returns value, so using else is unnecessary.